### PR TITLE
Add chrome to CI matrix

### PR DESCRIPTION
**Pull request by armanbilge**
Let's see what happens.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -123,13 +126,20 @@ lazy val unidoc213 = taskKey[Seq[File]]("Run unidoc but only on Scala 2.13")
 lazy val useFirefoxEnv =
   settingKey[Boolean]("Use headless Firefox (via geckodriver) for running tests")
 Global / useFirefoxEnv := false
+lazy val useChromeEnv =
+  settingKey[Boolean]("Use headless Chrome (via geckodriver) for running tests")
+Global / useChromeEnv := false
```
Comment (djspiewak) [inline build.sbt]: Random bikeshedding: would it be better if we had a `seleniumEnv = settingKey[Option[Env]]` where `Env` is either `Firefox` or `Chrome`?

File: build.sbt [hunk 2]
```diff
@@ -278,16 +292,21 @@ lazy val core = crossProject(JSPlatform, JVMPlatform)
       ProblemFilters.exclude[MissingClassProblem]("cats.effect.AsyncPropagateCancelation$"),
       // introduced by #1913, striped fiber callback hashtable, changes to package private code
       ProblemFilters.exclude[MissingClassProblem]("cats.effect.unsafe.FiberErrorHashtable"),
-      ProblemFilters.exclude[IncompatibleResultTypeProblem]("cats.effect.unsafe.IORuntime.fiberErrorCbs"),
+      ProblemFilters.exclude[IncompatibleResultTypeProblem](
+        "cats.effect.unsafe.IORuntime.fiberErrorCbs"),
```
Comment (djspiewak) [inline build.sbt]: Can we do the scalafmt in a separate commit?

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt]: Definitely!

File: build.sbt (hunk 2 above)
Comment (armanbilge) [reply build.sbt]: Oops, sure.

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt]: Using a `String` seemed easier than defining `Env` (I tried this and SBT balked, I think maybe we need to add a package declaration to `build.sbt`?) although I would certainly prefer the latter.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -54,6 +54,18 @@ jobs:
...
+          - ci: ciChrome
+            java: adopt@1.16
+          - ci: ciChrome
+            java: graalvm-ce-java8@21.1
+          - os: windows-latest
+            scala: 3.0.1
+          - os: windows-latest
+            scala: 2.12.14
+          - os: windows-latest
+            ci: ciChrome
```
Comment (vasilmkd) [inline .github/workflows/ci.yml:68]: I think we don't test Scala.js on Windows, so I vote for dropping these combinations.

File: build.sbt [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import org.openqa.selenium.remote.server.DriverFactory
```
Comment (vasilmkd) [inline build.sbt]: Can you please move this import below the license header and group it with the other ones.

File: build.sbt [hunk 5]
```diff
@@ -121,19 +128,36 @@ ThisBuild / githubWorkflowBuildMatrixExclusions ++= Seq(
 
 lazy val unidoc213 = taskKey[Seq[File]]("Run unidoc but only on Scala 2.13")
 
-lazy val useFirefoxEnv =
-  settingKey[Boolean]("Use headless Firefox (via geckodriver) for running tests")
-Global / useFirefoxEnv := false
+lazy val useBrowserEnv =
+  settingKey[Option[String]]("Use headless browser (via geckodriver) for running tests")
```
Comment (vasilmkd) [inline build.sbt]: `geckodriver` is the Firefox browser driver. Just drop the `(via geckodriver)` comment.

**Review by vasilmkd — CHANGES_REQUESTED** [review]
Looks good!

File: build.sbt (hunk 1 above)
Comment (vasilmkd) [reply build.sbt]: Maybe sbt needs a Scala source file in `project` which can define the `Env` trait and then it can be imported in the main `build.sbt`?

Comment (armanbilge) [reply build.sbt]: I can try that. Btw I think `Browser` is preferable to `Env`?

File: build.sbt (hunk 4 above)
Comment (armanbilge) [reply build.sbt]: Oops, yup. auto-import must have snuck that in there.

File: build.sbt (hunk 5 above)
Comment (armanbilge) [reply build.sbt]: Didn't realize this :) makes sense

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt]: Or if we do `Env` then we shouldn't make it an `Option` and instead include `NodeJS` as a possible `Env`?

Comment (vasilmkd) [reply build.sbt]: That sounds cool.

File: build.sbt (hunk 5 above)
Comment (vasilmkd) [reply build.sbt]: All good. 😄

**Review by djspiewak — DISMISSED** [review]
Very nice! Needs conflict resolution but otherwise this is awesome

**Edit** Oh it's already resolved. Okay perfect. :-D

File: build.sbt [hunk 6]
```diff
@@ -122,22 +131,37 @@ ThisBuild / githubWorkflowBuildMatrixExclusions ++= Seq(
...
-    options.setProfile(profile)
-    options.addArguments("-headless")
-    new SeleniumJSEnv(options)
-  } else {
-    old
+  useJSEnv.value match {
+    case NodeJS => old
+    case Firefox =>
+      val profile = new FirefoxProfile()
+      profile.setPreference("privacy.file_unique_origin", false)
```
Comment (armanbilge) [inline build.sbt:146]: We need to find the Chrome equivalent of this.

**Comment by armanbilge** [comment]
Sorry don't merge yet!! Need to get webworker going on Chrome.

File: build.sbt [hunk 7]
```diff
@@ -122,22 +131,38 @@ ThisBuild / githubWorkflowBuildMatrixExclusions ++= Seq(
...
+      val options = new FirefoxOptions()
+      options.setProfile(profile)
+      options.addArguments("-headless")
+      new SeleniumJSEnv(options)
+    case Chrome =>
+      val options = new ChromeOptions()
+      options.setHeadless(true)
+      options.addArguments("--allow-file-access-from-files")
```
Comment (armanbilge) [inline build.sbt:154]: that was easy!

**Comment by armanbilge** [comment]
Would you accept a PR disabling `sbt --client`?

File: build.sbt (hunk 6 above)
Comment (vasilmkd) [reply build.sbt:145]: https://dev.to/dengel29/loading-local-files-in-firefox-and-chrome-m9f

Comment (vasilmkd) [reply build.sbt:145]: https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/chromium/ChromiumOptions.html#addArguments(java.util.List)

**Review by vasilmkd — APPROVED** [review]

**Comment by vasilmkd** [comment]
Very cool work.

**Review by djspiewak — APPROVED** [review]

**Pull request merged**